src/v/redpanda/application.cc

```diff
@@ -661,14 +661,27 @@ void application::wire_up_services() {
           _proxy_client,
           to_yaml(*_proxy_client_config, config::redact_secrets::no))
           .get();
+        construct_single_service(_proxy_client_cache);
+
+        bool use_principal = _proxy_config->authn_type()
```

Comment:
This should not check the proxy config but instead check when sasl is enabled on the Kafka API.
In the Kafka API, we support no auth, SASL, or mTLS auth. Only SASL requires a principal on kafka clients (e.g., the pandaproxy clients).